Restructuring and addition to permissions docs (#3270)
hashicorp/boundary: website/content/docs/concepts/security/permissions/index.mdx

## Overview

> Boundary's permissions model is a composable, RBAC, allow-only model that

Author: Received feedback that this statement may be confusing, especially for an opening sentence in an overview, will edit

attempts to marry flexibility with usability. This section discusses the permission
model's fundamental concepts, provides examples grant strings, and contains a table 

> that acts as an cheat sheet to help those new to its grant syntax with crafting roles.

Reviewer: ```suggestion
that acts as a cheat sheet to help those new to its grant syntax with crafting roles.
```

Boundary's [domain model](/boundary/docs/concepts/domain-model) is based on resource
types. These can be implemented directly, such as with targets, or they can be
abstract types that are implemented by concrete types within the system. As an
example of the latter, a host catalog is an abstract type and a Static host
catalog is a concrete type.

At a very high level, permissions within Boundary are declared via grant strings
and mapped to users via roles.

> ## Grant Strings

Reviewer: ```suggestion
## Grant strings
```

A [grant string](/boundary/docs/concepts/security/permissions/grants) is a mapping 
that describes a resource or set of resources and the permissions that should be 
granted on them. Grant strings can be composed of up to 4 parts: 
- ID
- Type
- Actions 

> - Output Fields

Reviewer: ```suggestion
- Output fields
```

To learn about writing grant strings with examples, see [Grant Strings](/boundary/docs/concepts/security/permissions/grants).

## Roles

A [role](/boundary/docs/concepts/domain-model/roles) maps grant strings to principals (users and groups). 
Every role assigns grants within a **specific scope**: either the scope in which the role
exists, or a scope that is a child of the scope in which the role exists,

> controlled by the role's "grant scope ID" value.

Reviewer: ```suggestion
as determined by the role's "grant scope ID" value.
```

> When a request is made, the scope in which to discover grants is either provided by the client 

Reviewer: ```suggestion
When a user makes a. request, the scope in which to discover grants is either provided by the client 
```

  role

Roles are composable; a user's final set of grants will be composed of grants
that originate from all matching roles.

> To learn more about Boundary's access management system, see IAM [LINK TBD].

Reviewer: Reminder to replace this placeholder link.
